receiver/prometheusreceiver/internal/starttimemetricadjuster.go

```diff
@@ -19,21 +20,32 @@ var (
 
 type startTimeMetricAdjuster struct {
 	startTimeMetricRegex *regexp.Regexp
+	fallbackStartTime    *time.Time
 	logger               *zap.Logger
 }
 
 // NewStartTimeMetricAdjuster returns a new MetricsAdjuster that adjust metrics' start times based on a start time metric.
-func NewStartTimeMetricAdjuster(logger *zap.Logger, startTimeMetricRegex *regexp.Regexp) MetricsAdjuster {
+func NewStartTimeMetricAdjuster(logger *zap.Logger, startTimeMetricRegex *regexp.Regexp, useCollectorStartTimeFallback bool) MetricsAdjuster {
+	var fallbackStartTime *time.Time
+	if useCollectorStartTimeFallback {
+		now := time.Now()
+		fallbackStartTime = &now
+	}
 	return &startTimeMetricAdjuster{
 		startTimeMetricRegex: startTimeMetricRegex,
+		fallbackStartTime:    fallbackStartTime,
 		logger:               logger,
 	}
 }
 
 func (stma *startTimeMetricAdjuster) AdjustMetrics(metrics pmetric.Metrics) error {
 	startTime, err := stma.getStartTime(metrics)
 	if err != nil {
-		return err
+		if stma.fallbackStartTime == nil {
+			return err
+		}
+		stma.logger.Warn("Couldn't get start time for metrics. Using fallback start time.", zap.Error(err))
```

Comment:
Does this need to be at the `Warn` level?  Wouldn't this be a fairly high-rate log entry if none of the processed metrics have a start time?